packages/lean-imt/src/lean-imt.ts

```diff
@@ -233,6 +233,59 @@ export default class LeanIMT<N = bigint> {
         this._nodes[this.depth] = [node]
     }
 
+    /**
+     * Updates m leaves all at once.
+     * It is more efficient than using the {@link LeanIMT#update} method m times because it
+     * prevents updating middle nodes several times. This would happen when updating leaves
+     * with common ancestors. The naive approach of calling 'update' m times has complexity
+     * O(m*log(n)) (where n is the number of leaves of the tree), which ends up in
+     * O(n*log(n)) when m ~ n. With this new approach, this ends up being O(n) because every
+     * node is updated at most once and there are < 2*n nodes in the tree.
```

Comment:
```suggestion
     * node is updated at most once and there are around 2*n nodes in the tree.
```

I would say that there are around or about 2*n elements because for a LeanIMT with 10 leaves, there are 21 nodes and a tree with 6 leaves has 12 nodes. 

I think that a tree has < 2*n nodes is only true when n is a power of 2 because the tree would be a perfect binary tree. A perfect binary tree would have `2^(d+1)-1` which is `2n-1` which is `<2n`.

For the LeanIMT I think the answer would be `< 2n + log(n) + 1` but I think it's fine to say around 2n because a linear function grows faster than a logarithmic one. Saying `< 3n` or `< 4n` would also be fine.

Reply:
You're 100% right. I was only thinking about the perfect tree case. Fixed.